pkg/cmd/secret/list/list.go

```diff
@@ -162,6 +173,65 @@ func getOrgSecrets(client *api.Client, orgName string) ([]*Secret, error) {
 	return secrets, nil
 }
 
+func getEnvSecrets(client *api.Client, repo ghrepo.Interface, repoGetter func(client *api.Client, repo ghrepo.Interface) (*api.Repository, error), envName string) ([]*Secret, error) {
+	host := ghinstance.OverridableDefault()
+
+	repoId, err := getRepoId(client, repo, repoGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	secrets, err := getSecrets(client, host, fmt.Sprintf("repositories/%s/environments/%s/secrets", repoId, envName))
+	if err != nil {
+		return nil, err
+	}
+
+	type responseData struct {
+		TotalCount int `json:"total_count"`
+	}
+
+	for _, secret := range secrets {
+		if secret.SelectedReposURL == "" {
+			continue
+		}
+		u, err := url.Parse(secret.SelectedReposURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed determining selected repositories for %s: %w", secret.Name, err)
+		}
+
+		var result responseData
+		err = client.REST(u.Host, "GET", u.Path[1:], nil, &result)
+		if err != nil {
+			return nil, fmt.Errorf("failed determining selected repositories for %s: %w", secret.Name, err)
+		}
+		secret.NumSelectedRepos = result.TotalCount
+	}
```

Comment:
I think that there is no concept of "selected repos" for an environment secret (only for org secrets), so this whole section can go away.

Reply:
I love deleting code. Thanks!